components/playground/playground.go

```diff
@@ -785,68 +786,82 @@ func (p *Playground) bootCluster(ctx context.Context, env *environment.Environme
 	p.booted = true
 
 	var succ []string
-	for _, db := range p.tidbs {
-		prefix := color.YellowString("Waiting for tidb %s ready ", db.Addr())
-		bar := progress.NewSingleBar(prefix)
-		bar.StartRenderLoop()
-		if s := checkDB(db.Addr()); s {
-			succ = append(succ, db.Addr())
-			bar.UpdateDisplay(&progress.DisplayProps{
-				Prefix: prefix,
-				Mode:   progress.ModeDone,
-			})
-		} else {
-			bar.UpdateDisplay(&progress.DisplayProps{
-				Prefix: prefix,
-				Mode:   progress.ModeError,
-			})
+	if len(p.tidbs) > 0 {
+		var wg sync.WaitGroup
+		bars := progress.NewMultiBar(color.YellowString("Waiting for tidb instances ready\n"))
+		for _, db := range p.tidbs {
+			wg.Add(1)
+			prefix := color.YellowString(db.Addr())
+			bar := bars.AddBar(prefix)
+			go func(dbInst *instance.TiDBInstance) {
+				defer wg.Done()
+				if s := checkDB(dbInst.Addr(), options.tidb.UpTimeout); s {
+					succ = append(succ, dbInst.Addr())
+					bar.UpdateDisplay(&progress.DisplayProps{
+						Prefix: prefix,
+						Mode:   progress.ModeDone,
+					})
+				} else {
+					bar.UpdateDisplay(&progress.DisplayProps{
+						Prefix: prefix,
+						Mode:   progress.ModeError,
+					})
+				}
+			}(db)
 		}
-		bar.StopRenderLoop()
+		bars.StartRenderLoop()
+		wg.Wait()
+		bars.StopRenderLoop()
 	}
 
 	if len(succ) > 0 {
 		// start TiFlash after at least one TiDB is up.
-		startTiFlash := func() error {
+		var started []*instance.TiFlashInstance
+		for _, flash := range p.tiflashs {
+			if err := p.startInstance(ctx, flash); err != nil {
+				fmt.Println(color.RedString("TiFlash %s failed to start: %s", flash.Addr(), err))
+			} else {
+				started = append(started, flash)
+			}
+		}
+		p.tiflashs = started
```

Comment:
ignore tiflash instances that fail to start like how master branch works. 
however, I think it's better to exit and throw error when fail to start tiflash just like other components.

Reply:
I think it's because TiFlash failed to start is not a critical issue, the cluster can still bootstrap. So it's better to ignore it and just print out a log to let the user know what happened.

Reply:
yeah it's reasonable. But IMO it is based on the fact that the number of tiflash instances defaults to 1 rather than 0 while some user actually don't intend to use tiflash. 
BTW, some days ago, I've found that both en and zh documentations say the number of tiflash instances defaults to 0. I wonder if it's neccessary to set the default number to 1.

Reply:
I think we should update the document. In early releases, it's 0. But we changed it to 1 in recent releases.